service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -28,69 +35,198 @@
 
     private static final Mustache SPECIMEN_COMPOUND_STATEMENT_TEMPLATE =
         TemplateUtils.loadTemplate("specimen_compound_statement_template.mustache");
-    private static final String EFFECTIVE_TIME_TEMPLATE = "<effectiveTime><center value=\"%s\"></effectiveTime>";
+    private static final String FASTING_STATUS_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect"
+        + "-FastingStatus-1";
+    private static final String EFFECTIVE_TIME_CENTER_TEMPLATE = "<center value=\"%s\"/>";
+    private static final String DEFAULT_TIME_VALUE = "<center nullFlavor=\"UNK\"/>";
 
     private final MessageContext messageContext;
     private final ObservationMapper observationMapper;
 
-    public String mapSpecimenToCompoundStatement(Specimen specimen, List<Observation> observations, DiagnosticReport diagnosticReport) {
+    public String mapSpecimenToCompoundStatement(Specimen specimen, List<Observation> observations, String diagnosticReportIssuedDate) {
+        String mappedObservations = mapObservationsAssociatedWithSpecimen(specimen, observations);
+
+        var specimenCompoundStatementTemplateParameters = SpecimenCompoundStatementTemplateParameters.builder()
+            .compoundStatementId(messageContext.getIdMapper().getOrNew(ResourceType.Specimen, specimen.getId()))
+            .diagnosticReportIssuedDate(diagnosticReportIssuedDate)
+            .effectiveTime(prepareEffectiveTimeForSpecimen(specimen))
+            .observations(mappedObservations);
+
+        buildType(specimen).ifPresent(specimenCompoundStatementTemplateParameters::type);
+        buildAccessionIdentifier(specimen).ifPresent(specimenCompoundStatementTemplateParameters::accessionIdentifier);
+        buildPertinentInformation(specimen).ifPresent(specimenCompoundStatementTemplateParameters::pertinentInformation);
+        buildParticipant(specimen).ifPresent(specimenCompoundStatementTemplateParameters::participant);
+
+        return TemplateUtils.fillTemplate(
+            SPECIMEN_COMPOUND_STATEMENT_TEMPLATE,
+            specimenCompoundStatementTemplateParameters.build()
+        );
+    }
+
+    public static String prepareEffectiveTimeForSpecimen(Specimen specimen) {
+        Optional<DateTimeType> effectiveDate = getCollectionDateTime(specimen);
+
+        return effectiveDate.map(date -> String.format(EFFECTIVE_TIME_CENTER_TEMPLATE, DateFormatUtil.toHl7Format(date)))
+            .orElse(DEFAULT_TIME_VALUE);
+    }
+
+    private static Optional<DateTimeType> getCollectionDateTime(Specimen specimen) {
+        Optional<DateTimeType> effectiveDate = Optional.empty();
+
+        if (specimen.hasReceivedTime()) {
+            effectiveDate = Optional.of(specimen.getReceivedTimeElement());
+        }
```

Comment:
As this is overwritten if the specimen has a collection, it could be the `else` of that `if`

Reply:
No, we cannot do it this way because the order of date it:

1 collection.time or collection.period
2 receivedTime
3 nothing

So if we have collection time any of them we should not consider receivingTime, and to avoid extra ifs I changed the order.

Reply:
Ah, didn't notice that the last one was `else if` - so we can pass through that second `if` without changing it after all. Gotcha!

Reply:
yes, maybe I should refactor it to be more clear. I wanted to avoid ifs but it is started to be catchy.

Reply:
I changed it a little, hope it is more understandable now.